11.15.2 react-native TS errors

## 🐛 Bug Report

Hello, we've been experiencing a problem with the new version that was released today. It fails our builds with the following error:
```
../node_modules/react-i18next/ts4.1/index.d.ts(11,27): error TS2307: Cannot find module 'react-native' or its corresponding type declarations.
../node_modules/react-i18next/ts4.1/index.d.ts(242,3): error TS2706: Required type parameters may not follow optional type parameters.
```

## To Reproduce

I'm not sure if it's our dependencies or a bug in the new types that were added for the version. For now, we locked the version on 11.15.1 and it solved the issue

## Expected behavior
New version doesn't  break builds

## Your Environment
node 16
typescript 4.5.4
react-i18next@11.15.2
i18next@20.4.0
*os*: Linux bitbucket pipeline
- also fails locally on MacOS 12.0.1 (21A559)

Also experiencing this issue. With a bit more of the error log pasted, it seems like the ts4.1/index.d.ts is expecting react-native to always be available. 
```
node_modules/react-i18next/ts4.1/index.d.ts:11:27 - error TS2307: Cannot find module 'react-native' or its corresponding type declarations.

11 import { TextProps } from 'react-native';
                             ~~~~~~~~~~~~~~

node_modules/react-i18next/ts4.1/index.d.ts:242:3 - error TS2706: Required type parameters may not follow optional type parameters.
242   E extends React.HTMLProps<HTMLDivElement> | TextProps
```

Looking at the diff between 15.1 => 15.2 it seems like `@types/react-native` was pulled in as a [dev dependency]( https://github.com/i18next/react-i18next/compare/v11.15.1...v11.15.2#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R52) but appears to be required for the [index.d.ts file](https://github.com/i18next/react-i18next/compare/v11.15.1...v11.15.2#diff-61782f9118a033e04ba53dd09fb1e38d10dc39d2dfb9f8474f3dad935f16ea21R11)

--- a/package.json
+++ b/package.json
@@ -49,7 +49,6 @@
     "@testing-library/jest-dom": "^5.11.6",
     "@testing-library/react": "^11.2.7",
     "@testing-library/react-hooks": "^3.4.2",
-    "@types/react-native": "^0.66.6",
     "all-contributors-cli": "^6.20.0",
     "babel-core": "^7.0.0-bridge.0",
     "babel-eslint": "^10.0.3",

--- a/ts4.1/index.d.ts
+++ b/ts4.1/index.d.ts
@@ -8,7 +8,6 @@ import i18next, {
   TFunctionResult,
 } from 'i18next';
 import * as React from 'react';
-import { TextProps } from 'react-native';
 
 type Subtract<T extends K, K> = Omit<T, keyof K>;
 
@@ -239,7 +238,7 @@ export type TransProps<
   K extends TFuncKey<N, TKPrefix> extends infer A ? A : never,
   N extends Namespace = DefaultNamespace,
   TKPrefix = undefined,
-  E extends React.HTMLProps<HTMLDivElement> | TextProps
+  E = React.HTMLProps<HTMLDivElement>
 > = E & {
   children?: React.ReactNode;
   components?: readonly React.ReactNode[] | { readonly [tagName: string]: React.ReactNode };
@@ -259,7 +258,7 @@ export function Trans<
   K extends TFuncKey<N, TKPrefix> extends infer A ? A : never,
   N extends Namespace = DefaultNamespace,
   TKPrefix extends KeyPrefix<N> = undefined,
-  E extends React.HTMLProps<HTMLDivElement> | TextProps = React.HTMLProps<HTMLDivElement>
+  E = React.HTMLProps<HTMLDivElement>
 >(props: TransProps<K, N, TKPrefix, E>): React.ReactElement;
 
 export function useSSR(initialI18nStore: Resource, initialLanguage: string): void;

